pennylane/transforms/hamiltonian_expand.py

```diff
@@ -111,10 +111,7 @@ def naive_processing_fn(res, coeffs, offset):
     """
     dot_products = []
     for c, r in zip(coeffs, res):
-        if qml.math.ndim(c) == 0 and qml.math.size(r) != 1:
-            dot_products.append(qml.math.squeeze(r) * c)
-        else:
-            dot_products.append(qml.math.dot(qml.math.squeeze(r), c))
+        dot_products.append(qml.math.dot(qml.math.squeeze(r), c))
```

Comment:
Could lines 51-54 of this file be reduced in the same way? :thinking:

Reply:
I tried, doesn't seem like it works. Something else goes wrong if I simplify that part.